lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/auth/OAuthRefreshTokenDaoImpl.java

```diff
@@ -85,6 +87,25 @@
 			+ " WHERE " + COL_OAUTH_REFRESH_TOKEN_PRINCIPAL_ID + " = :" + PARAM_PRINCIPAL_ID
 			+ " AND " + COL_OAUTH_REFRESH_TOKEN_CLIENT_ID + " = :" + PARAM_CLIENT_ID;
 
+	/*
+	 * We use a JOIN because
+	 *   - MySQL doesn't support OFFSET when using DELETE
+	 *   - MySQL doesn't support LIMIT & IN/ALL/ANY/SOME subquery (so we can't do DELETE WHERE id IN (SELECT ... LIMIT x OFFSET y))
+	 *
+	 * https://stackoverflow.com/a/42030157
+	 */
+	private static final String DELETE_LEAST_RECENTLY_USED_ACTIVE_TOKEN = "DELETE t FROM " + TABLE_OAUTH_REFRESH_TOKEN + " t "
+			+ " JOIN ("
+				+ "SELECT tt." + COL_OAUTH_REFRESH_TOKEN_ID
+				+ " FROM " + TABLE_OAUTH_REFRESH_TOKEN + " tt "
+				+ " WHERE " + COL_OAUTH_REFRESH_TOKEN_PRINCIPAL_ID + " = :" + PARAM_PRINCIPAL_ID
+				+ " AND " + COL_OAUTH_REFRESH_TOKEN_CLIENT_ID + " = :" + PARAM_CLIENT_ID
+				+ " AND " + COL_OAUTH_REFRESH_TOKEN_LAST_USED
```

Comment:
I don't think you need AND COL_OAUTH_REFRESH_TOKEN_LAST_USED > (NOW() - INTERVAL : PARAM_DAYS DAY).  If you omit it then this SQL guarantees that it never leaves more than PARAM_MAX_NUM_TOKENS for a user+client.  If you include it then while you clean up excessive, unexpired tokens, any expired ones will just linger forever.